Backend/src/controllers/cluster.controller.ts

```diff
@@ -0,0 +1,113 @@
+import { Request, Response } from 'express';
+
+import { Cluster, ClusterInput } from '../models/cluster.model';
+import { Token } from '../models/token.model';
+import { User } from '../models/user.model';
+
+const addNewCluster = async (req: Request, res: Response) => {
+  const { name, description, host, port, ownerID } = req.body;
+  const accessToken = req.headers.authorization?.split(' ')[1];
+  try {
+    const token = await Token.findOne({ accessToken });
+    if (!token) {
+      return res.status(401).json({ message: 'Unauthorized' });
+    }
+    const user = await User.findOne({ _id: token.userId });
+    if (!user) {
+      return res.status(404).json({ message: 'User not found' });
+    }
+
+    const newCluster: ClusterInput = {
+      name,
+      description,
+      host,
+      port,
+      userIDs: [],
+      clusterOwner: token.userId,
+    };
+
+    const clusterCreated = await Cluster.create(newCluster);
+    
+    return res.status(201).json({ data: clusterCreated });
+  } catch (err) {
+    return res.status(500).send('Server error');
```

Comment:
We should not return 500 error to the UI. User is not able to take action on 500 error. Rather this needs to be logged at the server side and user should get meaningful error message.

Reply:
okay